cdap-tms/src/main/java/io/cdap/cdap/messaging/client/ClientMessagingService.java

```diff
@@ -92,17 +92,17 @@
  *
  * NOTE: This class shouldn't expose to end user (e.g. cdap-client module).
  */
-public final class ClientMessagingService implements MessagingService {
+public class ClientMessagingService implements MessagingService {
 
-  private static final HttpRequestConfig HTTP_REQUEST_CONFIG = new DefaultHttpRequestConfig(false);
   private static final TransactionCodec TRANSACTION_CODEC = new TransactionCodec();
   private static final Gson GSON = new Gson();
   // These types for only for Gson to use, hence using the gson TypeToken instead of guava one
   private static final Type TOPIC_PROPERTY_TYPE = new TypeToken<Map<String, String>>() { }.getType();
   private static final Type TOPIC_LIST_TYPE = new TypeToken<List<String>>() { }.getType();
+  protected static final HttpRequestConfig HTTP_REQUEST_CONFIG = new DefaultHttpRequestConfig(false);
 
-  private final RemoteClient remoteClient;
-  private final boolean compressPayload;
+  protected RemoteClient remoteClient;
+  protected boolean compressPayload;
```

Comment:
should keep these private final, the Local one should call the constructor from here.